Sources/GutenbergKit/Sources/EditorViewControllerDelegate.swift

```diff
@@ -20,9 +20,94 @@ public protocol EditorViewControllerDelegate: AnyObject {
     /// - warning: This is currently also called for the initial render, which
     /// is probably not how it should be in the production design.
     func editor(_ viewController: EditorViewController, didUpdateContentWithState state: EditorState)
+
+    func editor(_ viewController: EditorViewController, didRequestMediaFromSiteMediaLibrary config: OpenMediaLibrary)
 }
 
 public struct EditorState {
     /// Set to `true` if the editor has non-empty content.
     public var isEmpty = true
 }
+
+public struct OpenMediaLibrary: Decodable {
+    public let allowedTypes: [MediaType]
+    public let multiple: Bool
+    public let value: Value?
+
+    private enum CodingKeys: String, CodingKey {
+        case allowedTypes, multiple, value
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        multiple = try container.decode(Bool.self, forKey: .multiple)
+
+        // Decode allowedTypes as [String] and convert to [MediaType]
+        // If allowedTypes is not present in the JSON, use an empty array
+        let stringTypes = try container.decodeIfPresent([String].self, forKey: .allowedTypes) ?? []
+        allowedTypes = stringTypes.map { MediaType(fromJSString: $0) }
+
+        // Decode value as either Int? or [Int]?
+        if let singleValue = try? container.decodeIfPresent(Int.self, forKey: .value) {
+            value = .single(singleValue)
+        } else if let multipleValues = try? container.decodeIfPresent([Int].self, forKey: .value) {
+            value = .multiple(multipleValues)
+        } else {
+            value = nil
+        }
+    }
+
+    public enum MediaType: String, Decodable {
```

Comment:
Meant to note that the `MediaType` concept was copied and adapted from [Gutenberg Mobile](https://github.com/WordPress/gutenberg/blob/5432049fcaacd4a626ea8ff981520515e29a5fae/packages/react-native-bridge/ios/Gutenberg.swift#L315-L329) as well.